### PR TITLE
Added updated notebook with testing instructions

### PR DESCRIPTION
**Pull request by franflame**
Includes an updated version of Search & Download with minor changes to error messaging and modified column names specific to our new selected dataset. 

Also includes instructions for running the tests locally with nbval, with documentation of known issues.

### CODE REVIEW

File: demo-notebooks/testing_instructions.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+```pip install nbval```
+
+After confirming that the testing framework has been installed, you need to provide a valid token from your ESS-DIVE account (new one needed every 24 hours):
+
+- Open "secrets.txt": enter your authentication token and save the file 
+
+NOTE: Authentication tokens for the same account vary between the specific site used (e.g. ess-dive production site vs. data-stage)
+Please ensure that you are retrieving the token from the correct site.
+"Main_Site_Search_&_Download-Python-new-testable-example.ipynb" uses the token from the ess-dive production site
+"Data-Stage_Search_&_Download-Python-new-testable-example.ipynb" uses the token from data-stage.ess-dive.lbl.gov
```
Comment (mburrus) [inline demo-notebooks/testing_instructions.txt:21]: Remove line 21 if this file will not be included in the PR: "Data-Stage_Search_&_Download-Python-new-testable-example.ipynb"

File: demo-notebooks/testing_instructions.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
+A tested version of Search & Download has been created, named "Main_Site_Search_&_Download-Python-new-testable-example.ipynb"
+Modifications were made to handle our example dataset and improve error messages, but the core features of the notebook remain unchanged.
```
Comment (mburrus) [inline demo-notebooks/testing_instructions.txt:2]: I recommend adding the date that you copied "Search & Download - Python.ipynb" or the version of that notebook you copied.

File: demo-notebooks/testing_instructions.txt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+then 
+
+```pip install nbval```
+
+After confirming that the testing framework has been installed, you need to provide a valid token from your ESS-DIVE account (new one needed every 24 hours):
+
+- Open "secrets.txt": enter your authentication token and save the file 
+
+NOTE: Authentication tokens for the same account vary between the specific site used (e.g. ess-dive production site vs. data-stage)
```
Comment (mburrus) [inline demo-notebooks/testing_instructions.txt:18]: If the data-stage notebook is removed from the PR, consider combining lines 18-20 into something like this:
> NOTE: The testing notebook ("Main_Site_Search_&_Download-Python-new-testable-example.ipynb") points to our production endpoint at `https://data.ess-dive.lbl.gov/`. Please ensure that you retrieve your authentication token from the correct endpoint.

File: demo-notebooks/testing_instructions.txt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Known issues: 
+
+- If the notebook does not receive a correct response after the API call (in Cell 4), most cells in the notebook will also fail (typically 22 failed, 9 passed). 
+There are various causes to this problem; known ones include: an invalid token, ESS-DIVE being under maintenance, and if the search terms do not correspond to a valid dataset.
+
+- Cells 20 and onwards will fail if there is no CSV file found in the dataset. Note that the default example in the current version of Search & Download does not have a CSV file.
+
+- Cells 23 and onwards uses column names from the specific CSV file that is being opened. It will cause an error if the column names in the cells do not match up with those in the CSV file
+In both "Search_&_Download-Python-new-testable-example.ipynb" and "Data-Stage_Search_&_Download-Python-new-testable-example.ipynb,"
```
Comment (mburrus) [inline demo-notebooks/testing_instructions.txt:38]: May need to remove mention of the data-stage notebook if it is not included in the PR. Also the main-site notebook name is inconsistent here.

**Review by mburrus — COMMENTED** [review]
@franflame  
1. Is it possible to rename the main-site notebook now that it's no longer necessary to have a distinction between the main-site and data-stage notebooks? If it will require refactoring your code then maybe it's not worth it.

2. Also I ran the test in PyCharm terminal. Didn't you use some program during your demos that let you visually see the test run through the notebook? If so, what was that program and can that be added to the top of the instructions or a readme as a suggested program to run the tests on?

@vchendrix @shreddd  I'm concerned about how the testing files will appear to users who are just here to download the demo notebooks. Am I understanding correctly that the testing notebooks were designed just for us admin to use before we push a new version of our demo notebooks?
1. I'd like to at least add a readme to the demo-notebooks directory as a part of this PR to explain which files are used for testing, that they're just used by ESS-DIVE admin and should be ignored by regular users.
2. Is there a better way to organize this directory to distinguish between the demo notebooks we offer and the tests? Like adding a new folder.

File: demo-notebooks/testing_instructions.txt (hunk 1 above)
Comment (vchendrix) [reply demo-notebooks/testing_instructions.txt:21]: @mburrus These are great comments.  let's merge @franflame code as-is and then someone from our team can take if from there and make it ready for public use.

Comment (vchendrix) [reply demo-notebooks/testing_instructions.txt:21]: @mburrus I take back my previous comment.  This is a great point.

**Comment by vchendrix** [comment]
>  @vchendrix @shreddd I'm concerned about how the testing files will appear to users who are just here to download the demo notebooks. Am I understanding correctly that the testing notebooks were designed just for us admin to use before we push a new version of our demo notebooks?
> 
> I'd like to at least add a readme to the demo-notebooks directory as a part of this PR to explain which files are used for testing, that they're just used by ESS-DIVE admin and should be ignored by regular users.
> Is there a better way to organize this directory to distinguish between the demo notebooks we offer and the tests? Like adding a new folder.

@mburrus I will let @shreddd say what he has in his mind and far as the intention.  Eventually, we will want these tests automated in Github actions. However, until then they will need to ba manually run.  We can take was @franflame has done and reshape it to how we think it will work best.
